rpc/RPCSession.java

```diff
@@ -54,8 +54,8 @@
 
         sessionId = blockingGrpcStub.sessionOpen(openReq).getSessionId();
         pulse = new Timer();
-        pulse.scheduleAtFixedRate(this.new PulseTask(), 0, 5000);
         isOpen = new AtomicBoolean(true);
+        pulse.scheduleAtFixedRate(this.new PulseTask(), 0, 5000);
```

Comment:
ah yes, another reminder that running new threads in the constructor is evil...

Reply:
I think this is fine. It's a little bit fragile but the solutions to that seem very arbitrary (e.g: initialising isOpen inline; using `scheduleAtFixedRate(this.new PulseTask(), 100, 5000)` etc....) so I don't think we can really do any better.